src/main/sensors/battery.c

```diff
@@ -35,77 +35,41 @@
 
 #include "fc/runtime_config.h"
 #include "fc/config.h"
-#include "fc/rc_controls.h"
 
 #include "io/beeper.h"
 
+#include "sensors/voltage.h"
+#include "sensors/current.h"
 #include "sensors/battery.h"
 
-
-// FIXME there is too much going on in here - the code is not re-usable and has lots of shared configuration, suggest splitting into these topics.
-// 1) voltage monitoring/adc conversion
-// 2) battery cell count detection and status (present or not)
-// 3) battery warnings (aka alarms)
-// 4) adc current sensor
-// 5) virtual current sensor (has a dependency on an RC control input)
-
 #define VBATT_PRESENT_THRESHOLD_MV    10
-#define VBATT_LPF_FREQ  1.0f
+
+uint16_t vbat;
 
 // Battery monitoring stuff
 uint8_t batteryCellCount = 3;       // cell count
 uint16_t batteryWarningVoltage;
```

Comment:
the problem there, is that the rest of the system uses it.  telemetry, blackbox, msp, etc.

I'm running out of time here, i think that change can wait for now.